### PR TITLE
 feat(inline-inputs): add labelWidth and inputWidth props (FE-4279)

### LINKED ISSUE #4300: Inline Input labels do not align with other inputs on a form
### Current behaviour

InlineInputs labels are always inline and the same width of the text within them, this does not match with the other fields you can put on a form which have properties to configure where to display them, how much width to give them, or when to break between displaying them inline or above the field.
The result is a strange looking form where the InlineInputs are not aligned with anything else, including each other.
image

### Expected behaviour

When I do not provide the labelInline property to an InlineInput component, the inline inputs span the full width of the form with the label displayed above them. If I do provide the labelInline property, then I can use the same labelWidth and input width properties so that it aligns with the other fields on the form.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/wonderful-moore-f1nc4?file=/src/index.js

### Suggested Solution

_No response_

### Carbon Version

80.1.0

### What browsers are you seeing the problem on?

Firefox

### What Operating System are you seeing the problem on?

Windows

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Inline inputs should have the ability to set label and content width the same way as other form elements 
Screenshot 2021-08-27 at 17 53 43

### Current behaviour
Inline inputs could not be aligned with other form elements when their labels are in-line
Screenshot 2021-08-27 at 17 53 30

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
~~- [ ] All themes are supported if required~~
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Additional context
Resolves #4300 

### Testing instructions
1. run npm start
2. open http://localhost:9001/?path=/story/design-system-form--default-with-sticky-footer
3. open http://localhost:9001/?path=/story/design-system-inline-inputs--default-story

### CODE REVIEW

File: src/components/form/form.stories.mdx [hunk 1]
```diff
@@ -50,25 +51,31 @@ When `stickyFooter` prop is set as true, footer becomes stickied to the bottom o
...
-      stickyFooter
-    >
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
-      <Textbox label="Textbox" />
+    <Form>
```
Comment (mkrds) [inline src/components/form/form.stories.mdx:54]: Why did you remove the footer buttons from `Form` ?

File: src/components/form/form.style.js [hunk 2]
```diff
@@ -53,6 +54,15 @@ export const StyledForm = styled.form`
       theme.spacing * fieldSpacing}px;
   }
 
+  ${StyledInlineInputs} {
```
Comment (mkrds) [inline src/components/form/form.style.js:57]: How about adding another `&` to raise specificity of the rule on line 51 instead of adding this whole block?

**Review by DipperTheDan — COMMENTED** [review]
Nothing further to add than what @mkrds has already commented. Address the comments and I'll happily approve 😄

File: src/components/form/form.style.js (hunk 2 above)
Comment (grabkowski) [reply src/components/form/form.style.js:57]: there's no need to do this, as it will specifically target inline inputs, I don't understand why should I raise specificity of it.

**Review by DipperTheDan — APPROVED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by nicktitchmarsh — APPROVED** [review]
